seacrowd/sea_datasets/qasina/qasina.py

```diff
@@ -0,0 +1,170 @@
+                    "context_id": datasets.Value("int32"),
+                    "context": datasets.Value("string"),
+                    "question_answers": datasets.Sequence({"type": datasets.Value("string"), "question": datasets.Value("string"), "answer": datasets.Value("string"), "answer_start": datasets.Value("int32"), "question_id": datasets.Value("int32")}),
+                    "context_length": datasets.Value("int32"),
+                    "context_title": datasets.Value("string"),
+                }
+            )
+
+        elif self.config.schema == f"seacrowd_{self.SEACROWD_SCHEMA_NAME}":
+            features = schemas.qa.features
+            features["meta"] = {"context_title": datasets.Value("string")}
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        urls = _URLS[_DATASETNAME]
+        filepath = dl_manager.download(urls)
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "filepath": filepath,
+                    "split": "train",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, filepath: Path, split: str) -> Tuple[int, Dict]:
+        with open(filepath) as file:
+            dataset = json.load(file)
+
+        if self.config.schema == "source":
+            for i, line in enumerate(dataset):
+                yield i, {
+                    "context_id": line["context_id"],
+                    "context": line["context"],
+                    "question_answers": [
+                        {
+                            "type": subline["type"],
+                            "question": subline["question"],
+                            "answer": subline["answer"],
+                            "answer_start": subline["answer_start"],
+                            "question_id": subline["question_id"],
+                        }
+                        for subline in line["question_answers"]
+                    ],
+                    "context_length": line["context_length"],
+                    "context_title": line["context_title"],
+                }
+
+        elif self.config.schema == f"seacrowd_{self.SEACROWD_SCHEMA_NAME}":
+            for line in dataset:
+                for question_answer in line["question_answers"]:
+                    id = question_answer["question_id"]
+
+                    yield id, {
+                        "id": id,
+                        "question_id": question_answer["question_id"],
+                        "document_id": line["context_id"],
+                        "question": question_answer["question"],
+                        "type": question_answer["type"],
```

Comment:
`"type": "extractive"`

Could you please move the current QA type value inside the `meta` variable? It would also be great to have the corresponding `context_length` and the `answer_start` inside the `meta` variable as well.

cc: What do you think, @MJonibek?

Reply:
I agree about the current QA type value to be added to meta, but I am not sure that `context_length` and `answer_start` are somehow needed

Reply:
Yeah, `context_length` is probably not needed, but `answer_start` is directly related to where the `answer` is found in the `context`.

Reply:
Oh, I see. Then it would be good to add current QA type and `answer_start` to the `meta`